app/models/manageiq/providers/kubevirt/infra_manager/provision/cloning.rb

```diff
@@ -54,4 +78,60 @@ def user_options(options)
     merged_options[:cloud_user_password] = ManageIQ::Password.try_decrypt(get_option(:root_password))
     merged_options.compact
   end
+
+  def values(template, options)
+    default_params = {}
+    template.parameters.each do |param|
+      name = param[:name].downcase
+      value = options[name.to_sym]
+      if value && name == "memory"
+        value = "#{value}Mi"
+      end
+
+      default_params[name] = value || param[:value]
+    end
+    default_params
+  end
+
+  def param_substitution!(object, params)
+    result = object
+    case result
+    when Hash
+      result.each do |k, v|
+        result[k] = param_substitution!(v, params)
+      end
```

Comment:
Can this be 

```suggestion
      result.transform_values! { |v| param_substitution!(v, params) }
```

Reply:
Sure

Reply:
Oh I didn't realize this came from elsewhere - it can be changed separately.